checkpoint/recovery.go

```diff
@@ -169,22 +168,35 @@ func RecoverWithDb(
 	if err != nil {
 		return nil, err
 	}
-	return recoverFromLocalFile(ctx, logger, db, localDB, fs, cfg, cpFile)
+	var newDB *sql.Database
+	createDb := func() (*sql.Database, error) {
+		var err error
+		newDB, err = sql.Open("file:" + filepath.Join(cfg.DataDir, cfg.DbFile))
+		return newDB, err
+	}
+	defer func() {
+		if newDB != nil {
+			newDB.Close()
+		}
+	}()
+
+	return RecoverFromLocalFile(ctx, logger, db, localDB, fs, cfg, cpFile, createDb)
 }
 
 type recoveryData struct {
 	accounts []*types.Account
 	atxs     []*atxs.CheckpointAtx
 }
 
-func recoverFromLocalFile(
+func RecoverFromLocalFile(
 	ctx context.Context,
 	logger *zap.Logger,
 	db *sql.Database,
 	localDB *localsql.Database,
 	fs afero.Fs,
 	cfg *RecoverConfig,
 	file string,
+	createDb func() (*sql.Database, error),
```

Comment:
This feels very hacky and as far as I can see is only needed for testing? The path to the created DB is part of the `RecoverConfig`. Why not set these values to where you want the DB to be created and open it afterwards to assert its contents?